derive-new/src/lib.rs

```diff
@@ -52,7 +52,38 @@ fn new_for_struct(ast: syn::MacroInput) -> quote::Tokens {
                 }
             }
         },
-        syn::Body::Struct(syn::VariantData::Tuple(_)) => panic!("#[derive(new)] cannot be used with tuple structs"),
+        syn::Body::Struct(syn::VariantData::Tuple(ref fields)) => {
+            // Special-case a single member - call the parameter "value" instead of "value1"
+            if fields.len() == 1 {
```

Comment:
I don't think it is worth special casing the single value case since the arg names are not really user-facing.